tqdm/utils.py

```diff
@@ -283,3 +283,13 @@ def _environ_cols_linux(fp):  # pragma: no cover
 
 def _term_move_up():  # pragma: no cover
     return '' if (os.name == 'nt') and (colorama is None) else '\x1b[A'
+
+
+try:
+    # TODO consider using wcswidth third-party package for 0-width characters
+    from unicodedata import east_asian_width
+except ImportError:
+    _text_width = len
+else:
+    def _text_width(s):
+        return len(s) + sum(east_asian_width(ch) in 'FW' for ch in _unicode(s))
```

Comment:
@FichteFoll @RedBug312 are you sure this is means to be `east_asian_width(ch) in 'FW'` rather than `east_asian_width(ch) == 'FW'` or something else?

Reply:
Half-width are the standard width for console output. We only need to check full-width and wide.
I'm not aware of wider glyphs.